mlx_steering_dump_parser.py

```diff
@@ -0,0 +1,204 @@
+                    dr_rec_type == dr_dump_rec_type.DR_DUMP_REC_TYPE_MATCHER_RX.value[0]:
+                dump_ctx.matcher.add_matcher_rx_tx(dr_obj)
+
+            elif dr_rec_type == dr_dump_rec_type.DR_DUMP_REC_TYPE_MATCHER_BUILDER.value[0]:
+                dump_ctx.matcher.add_builder(dr_obj)
+
+            # update Rule objects
+            elif dr_rec_type == dr_dump_rec_type.DR_DUMP_REC_TYPE_RULE.value[0]:
+                dump_ctx.rule = dr_obj
+                dump_ctx.matcher.add_rule(dr_obj)
+
+            elif dr_rec_type == dr_dump_rec_type.DR_DUMP_REC_TYPE_RULE_RX_ENTRY.value[0] \
+                    or dr_rec_type == dr_dump_rec_type.DR_DUMP_REC_TYPE_RULE_TX_ENTRY.value[0]:
+                dump_ctx.rule.add_rule_entry(dr_obj)
+
+            # update Action objects
+            elif dr_dump_rec_type.find_name(dr_rec_type).startswith('DR_DUMP_REC_TYPE_ACTION_'):
+                dump_ctx.rule.add_action(dr_obj)
+
+        dr_obj = None
+        if dump_ctx.domain:
+            dr_obj = dump_ctx.domain
+        elif dump_ctx.table:
+            dr_obj = dump_ctx.table.print_view
+        elif dump_ctx.matcher:
+            dr_obj = dump_ctx.matcher
+        elif dump_ctx.rule:
+            dr_obj = dump_ctx.rule
+
+        if view == dr_dump_view.DR_DUMP_VIEW_TREE:
+            dr_obj.print_tree_view(dump_ctx, verbose, raw)
+        else:
+            dr_obj.print_rule_view(dump_ctx, verbose, raw)
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+    description='''mlx_steering_dump.py - Steering dump tool''')
+    parser.add_argument('-f', dest="FILEPATH", default="", help='input steering dump file path')
+    parser.add_argument('-d', dest="trigger", default=[], metavar=('PID', 'PORT'), nargs=2, help='Trigger DPDK app to generate json dump file (-d <APP PID> <PORT NUMBER>)')
+    parser.add_argument('-t', action='store_true', default=False, dest='tree_view', help='tree view (default is rule view)')
+    parser.add_argument("-v", action="count", dest='verbose', default=0, help="increase output verbosity")
+    parser.add_argument('-r', action='store_true', default=False, dest='raw', help='raw output')
+    parser.add_argument('-version', action='store_true', default=False, dest='version', help='show version')
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+    if (args.version):
+            print_dr("Version %s\n" % g_version)
+            return 0
+
+    if (args.FILEPATH == ""):
+            print_dr("No input steering dump file provided (-f FILEPATH)\n")
+            return 0
+
+    if (len(args.trigger) == 2):
+            pid = int(args.trigger[0])
+            port = int(args.trigger[1])
+            dr_trigger.trigger_dump(pid, port, args.FILEPATH)
+
+    if (args.tree_view):
+        print_domain(args.FILEPATH, dr_dump_view.DR_DUMP_VIEW_TREE, args.verbose, args.raw)
+    else:
+        print_domain(args.FILEPATH, dr_dump_view.DR_DUMP_VIEW_RULE, args.verbose, args.raw)
```

Comment:
If trigger_dump fails, you will still continue to print_domain.
Need to have some status handling here.